src/modules/commander/Commander.cpp

```diff
@@ -2865,12 +2965,9 @@ Commander::set_main_state_rc()
 
 	/* Loiter switch overrides main switch */
 	if (_manual_control_switches.loiter_switch == manual_control_switches_s::SWITCH_POS_ON) {
-		res = main_state_transition(_status, commander_state_s::MAIN_STATE_AUTO_LOITER, _status_flags, &_internal_state);
+		res = try_mode_change(commander_state_s::MAIN_STATE_AUTO_LOITER, FlightModeChange::FallbackDisabled);
```

Comment:
Maybe this is a nitpick, but wouldn't it be simpler to just keep it separate?

 1. change main state directly, or
 2. try mode change with a suitable fallback selected for you

Then you don't need to litter try_mode_change with fallback checks and things are readable in place.

To avoid unnecessary string variations all over the place we can update print_reject_mode() to take the value (commander_state_s::MAIN_STATE_AUTO_LOITER). https://github.com/PX4/PX4-Autopilot/pull/17096

Reply:
@cmic0 how about this? https://github.com/PX4/PX4-Autopilot/pull/17265

Reply:
Good suggestion! I think when i initially started this PR i just kept things together to simplify the analysis as i was moving things around, this incremental piece you suggested makes it even cleaner!

I am going to cherry pick it with some minor changes directly here to speed up the process